### ISSUE TITLE
Add expansion parameter to backend spelldata API

### ISSUE THREAD

**Issue by emallson** [WoWAnalyzer/WoWAnalyzer#5145]
Example breakage in classic: https://i.imgur.com/3MkIDbc.png

This needs refinement to see the scope of the work. Unfamiliar with this part of the backend.

**Comment by jazminite** [comment]
Is this fixed? I think there's a few times I've seen Retail names chosen instead of Classic but haven't seen "Unknown spell" for awhile.

**Comment by emallson** [comment]
No, but passing spell objects instead of spell IDs skips the lookup and
just uses the bundled data so unknown spell issues should be way less
common now.

**Comment by emallson** [comment]
Which is to say: it would still be good to have this, but it's way less
important with other codebase changes we've done
